src/tasks/removedevice/taskremovedevice.cpp

```diff
@@ -49,6 +50,9 @@ void TaskRemoveDevice::run() {
   connect(request, &NetworkRequest::requestCompleted, this,
           [this](const QByteArray&) {
             logger.debug() << "Device removed";
+
+            emit MozillaVPN::instance()->recordGleanEventWithExtraKeys(
```

Comment:
I would move this into the VPN.deviceRemoved to be sure we don't forget similar removal calls.

We can pass an enum as extra param to the identify the source of the call.